build.sc

```diff
@@ -88,51 +88,78 @@ object emulator extends mill.Cross[Emulator](
   ("freechips.rocketchip.system.TestHarness", "freechips.rocketchip.system.MMIOPortOnlyConfig"),
   ("freechips.rocketchip.system.TestHarness", "freechips.rocketchip.system.CloneTileConfig"),
 )
-class Emulator(top: String, config: String) extends ScalaModule {
-  override def moduleDeps = Seq(rocketchip)
+class Emulator(top: String, config: String) extends Module {
 
-  override def scalaVersion: T[String] = T {
-    "2.13.10"
-  }
-
-  override def repositories = super.repositories ++ Seq(
-    MavenRepository("https://oss.sonatype.org/content/repositories/snapshots"),
-    MavenRepository("https://oss.sonatype.org/content/repositories/releases")
-  )
+  object generator extends ScalaModule {
```

Comment:
This doesn't need to be a `ScalaModule`

Reply:
The `Generator` below uses `moduleDeps`, `upstreamCompileOutput()` and `runClasspath()`, which are not provided by `Module`.

Reply:
It actually requires `rocketchip.upstreamCompileOutput()`, this Module is not a `ScalaModule` since it doesn't contain any Scala source codes.

Reply:
Actually I do agree split the elaborate specific codes into a standalone package.

Reply:
But this is not ready for now.
I think `generator` should only depend on the `rocketchip.jar` and `firtool`